src/code_e79b0_len_1920.c

```diff
@@ -64,41 +86,194 @@ s32 does_script_exist(s32 id) {
     return 0;
 }
 
-INCLUDE_ASM(code_e79b0_len_1920, does_script_exist_by_ref);
+s32 does_script_exist_by_ref(script_context* script) {
+    s32 i;
+
+    for(i=0; i < ARRAY_COUNT(gCurrentScriptListPtr); i++) {
+        if(script == (*gCurrentScriptListPtr)[i]) {
+            return 1;
+        }
+    }
+    return 0;
+}
 
-INCLUDE_ASM(code_e79b0_len_1920, set_script_priority);
+void set_script_priority(script_context* script, s8 priority) {
+    script->priority = priority;
+}
 
-INCLUDE_ASM(code_e79b0_len_1920, set_script_timescale);
+void set_script_timescale(script_context* script, f32 timeScale) {
+    script->timeScale = timeScale * gGlobalTimeSpace;
+}
 
 INCLUDE_ASM(code_e79b0_len_1920, set_global_timespace);
+/*
+// TODO: figure out why compiler/assembler isn't putting SWC1 in delay slot
+void set_global_timespace(f32 timeScale) {
+    //gGlobalTimeSpace = timeScale;
+    __asm__("LUI $at,0x802e;"
+        "SWC1 $f12,-0x6358($at);"
+    );
+}
+*/
+
 
 INCLUDE_ASM(code_e79b0_len_1920, get_global_timespace);
+/*
+// TODO: figure out why compiler/assembler isn't putting LWC1 in delay slot
+f32 get_global_timespace(void) {
+    //return gGlobalTimeSpace;
+    __asm__("LUI $at,0x802e;"
+        "LWC1 $f0,-0x6358($at);"
+    );
+}
+*/
 
-INCLUDE_ASM(code_e79b0_len_1920, set_script_group);
+void set_script_group(script_context* script, s8 groupFlags) {
+    script->groupFlags = groupFlags;
+}
 
 INCLUDE_ASM(code_e79b0_len_1920, bind_trigger);
 
 INCLUDE_ASM(code_e79b0_len_1920, bind_trigger_1);
 
-INCLUDE_ASM(code_e79b0_len_1920, suspend_group_script);
+void suspend_group_script(script_context* script, s32 groupFlags) {
+    int i;
+    script_context* scriptContextPtr;
+    script_context* childScript = script->childScript;
 
-INCLUDE_ASM(code_e79b0_len_1920, resume_group_script);
+    if (childScript != NULL) {
+        suspend_group_script(childScript, groupFlags);
+    }
 
-INCLUDE_ASM(code_e79b0_len_1920, suspend_all_script);
+    for(i=0; i < ARRAY_COUNT(gCurrentScriptListPtr); i++) {
+        scriptContextPtr = (*gCurrentScriptListPtr)[i];
+        if (scriptContextPtr != NULL && scriptContextPtr->parentScript == script) {
```

Comment:
`script == scriptContextPtr->parentScript` would read better if it doesn't affect codegen

Reply:
This one does affect codegen (mis-matches registers in the assembly otherwise)

Reply:
bummer, okay